expression/core/result.py

```diff
@@ -28,6 +28,7 @@
 
 from typing_extensions import TypeAlias, TypeGuard
 
+from . import option
```

Comment:
Not happy with option import result and result importing option. Can we use `if TYPE_CHECK)ING:` here to avoid getting circular dependencies in the running code?

Reply:
I'll take a look

Reply:
Thanks for your review. I've updated the PR with your suggestion.